File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>
using namespace std;

bool exist[21];

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int M;
    cin >> M;
    fill(exist, exist+21, false);
    
    for(int i=0; i<M; ++i) {
        string s;
        int x;
        cin >> s;



        if(s=="add") {
            cin >> x;
            
            exist[x] = true;
        } else if(s == "remove") {
            cin >> x;
            exist[x] = false;
        } else if(s == "check") {
            cin >> x;
            if(exist[x]) cout << 1 <<'\n';
            else cout << 0 << '\n';
        } else if(s == "toggle") {
            cin >> x;
            if(exist[x]) exist[x] = false;
            else exist[x] = true;
        } else if(s == "all") {
            fill(exist, exist+21, true);
        } else if(s == "empty") {
            fill(exist, exist+21, false);
        }
    }
    
    return 0;
}

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.*;


class Main {
  private static String S;
  private static int M;

  public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    M = Integer.parseInt(br.readLine());

    Set<Integer> set = new HashSet<>();
    for(int i=0; i<M;++i) {
      StringTokenizer st = new StringTokenizer(br.readLine());
      String a = st.nextToken();


      if(a.equals("add")) {
        int b = Integer.parseInt(st.nextToken());
        set.add(b);
      } else if(a.equals("remove")) {
        int b = Integer.parseInt(st.nextToken());
        set.remove(b);
      } else if(a.equals("check")) {
        int b = Integer.parseInt(st.nextToken());
        if(set.contains(b)) bw.write("1\n");
        else bw.write("0\n");
      } else if(a.equals("toggle")) {
        int b = Integer.parseInt(st.nextToken());
        if (set.contains(b)) {
          set.remove(b);
        } else {
          set.add(b);
        }
      } else if(a.equals("all")) {
        for(int k=1; k<=20; ++k) set.add(k);
      } else {
        set.clear();
      }
    }
    
    bw.flush();
  }
}